File: 3558/3558.c
int numberOfWaysToAssignEdgeWeight(struct TreeNode* root) {
    if (!root) return 0;
    int l = numberOfWaysToAssignEdgeWeight(root->left);
    int r = numberOfWaysToAssignEdgeWeight(root->right);
    return 1 + (l > r ? l : r);
}

File: 3558/3558.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var numberOfWaysToAssignEdgeWeight = function(root) {
    if (!root) return 0;
    return 1 + Math.max(numberOfWaysToAssignEdgeWeight(root.left), numberOfWaysToAssignEdgeWeight(root.right));
};